files/initramfs-tools/union-mount.j2

```diff
@@ -95,6 +103,63 @@ remove_not_in_allowlist_files()
     rm -f $allowlist_pattern_file
 }
 
+mount_docker_inram()
+{
+    if [ "$docker_inram_algo" = "tmpfs" ]; then
```

Comment:
@Staphylo could we split this PR to 2 PRs? One is for the docker_inram, the other is to turn on zram. With smaller PR, it would be easier to manage if any issue we want to revert or turn off.

Reply:
This PR is made of 2 commits, would you want these 2 commits as individual PRs?
 - fix docker_inram behavior on Arista systems
 - refactor docker mount to cleanly add zram support

Or would you rather see 3 commits (much longer due to dependencies):
 - fix docker_inram behavior on Arista systems
 - refactor union-mount.j2 with the introduction of `mount_docker` and only `tmpfs` support like it is today
 - add support for the zram compression?

Whatever split you decide, please bear in mind that these changes will take much longer to merge as all of these changes generate conflict with each other. So I'll have to open the PRs one by one and wait until the previous one is merged. Cherry-picking to older release branches, most of these PRs will conflict again.
I will also need to spend much more time getting builds and running testing on individual changes rather than testing the aggregated content.
I'll do as you prefer but this will take much longer for everything to merge and be available in 202012 and 202205.

Please let me know

Reply:
thanks for working on this. current two commits are ok for me. however, for current commits, i couldn't find the commit messages. we can do a rebase merge which keep your original commits. can you check if your original commits has proper commit messages?

Reply:
I updated the commits of this PR so they have the correct description.